### PR TITLE
Add fix for ios to enable passcode backup

### PR DESCRIPTION
**Pull request by ajnrajeev**
Added fix for ios to enable users to enter the
passcode in case the finger print authentication
fails. Otherwise, Enter passcode button is present
but no action is present on clicking the same

### CODE REVIEW

**Comment by NiklasMerz** [comment]
Thanks for submitting this feature. If I understand this correctly, you are checking for iOS 9. What happens if the iOS is higher then 9 but the device has no fingerprint scanner? Do you know how to support this. I will have a look this myself soon.

**Comment by ajnrajeev** [comment]
I am assuming that the availability check before the authenticate could fix this issue. What do you suggest?

**Comment by NiklasMerz** [comment]
Sorry I misunderstood the issue. Did you experienced crashes on iOS 8? `deviceOwnerAuthentication` was introduced in iOS 9 and on iOS 8 you had to use `deviceOwnerAuthenticationWithBiometrics`. So this fix makes the plugin usable on iOS 8? If you can confirm this, I could merge and release this soon.

**Comment by ajnrajeev** [comment]
My bad, I could have explained the issue more fully. I was using this plugin on an ios 10 device. And when the fingerprint is authenticated, it is all good. But when I use a non-matching print, the OS prompts me to use backup. When the policy is set to deviceOwnerAuthenticationWithBiometrics, and I choose to backup, nothing obviously happens. So I added the check to change the policy to deviceOwnerAuthentication on ios9+ devices, and the passcode screen came up. So you can use the biometrics in ios8 devices and finger print + passcode backup for ios9+ devices.

**Comment by NiklasMerz** [comment]
Thanks for explaining the issue. I have tested this on the simulator and an iPad and this works great. You can use this plugin even on devices without a TouchId sensor.

**Comment by NiklasMerz** [comment]
Maybe we should set this policy if the option "Use backup" is set like it is used for Android.

**Comment by ajnrajeev** [comment]
I can add more code to '**disableBackup**'. But this code would still be needed to make sure that even if disableBackup is not set for iOS8 devices, we still use the correct policy. This will make the policy dependent on OS version and backup as a feature for the plugin. I am doing one round of testing, and if you like to add both this fix and the disable backup option feature together, I can try to update this same pull request.
What are your thoughts?

**Comment by NiklasMerz** [comment]
I totally agree with you. If you could add this to the pull request this would be nice. I think this would be the best possible solution.

**Comment by ajnrajeev** [comment]
I have added these changes as well. Please do take a look

**Comment by NiklasMerz** [comment]
@ajnrajeev Would you please check my changes? This should fix the disable backup feature for iOS. If you think this ok, I will do a quick test and release a new version with your changes.

**Comment by ajnrajeev** [comment]
@NiklasMerz, I think you have changed the policy inside the _disable backup_ code. This might cause issues in iOS8 devices. 

I have tested my code in iOS10. When I set **disable backup** flag to **true**,  on wrong fingerprint entry, the native popup does not show the _Enter Passcode_ button. But once user inputs 3 wrong prints, the passcode backup pops up since I have set the biometrics policy for iOS > 8. This is as per expectations. When **disbale backup** is set to **false**, the native popup that comes when the user enters wrong print, has by default the _Enter Passcode_ button. On tapping this the passcode screen is displayed. In iOS 8 devices, if the policy is set to biometrics, the device may crash. So do we really need to add that line of code inside _disable backup_?

 I liked the _localizedFallbackTitle_ changes though.

**Comment by ajnrajeev** [comment]
One more input I have is, when you are setting the localized title, we can add that in the else part of disable backup. This will avoid issues when user sets a localized title and sets disable backup as true. What do you think?

**Comment by NiklasMerz** [comment]
@ajnrajeev Thank you very very much. I totally agree you and have done the changes. It is always better to get a second opinion :+1: .

File: src/ios/Fingerprint.swift [hunk 1]
```diff
@@ -23,13 +23,24 @@ import LocalAuthentication
...
+        if disableBackup {
+            authenticationContext.localizedFallbackTitle = "";
+            policy = .deviceOwnerAuthenticationWithBiometrics;
+        }else{
+          if let localizedFallbackTitle = data?["localizedFallbackTitle"] as! String? {
+            authenticationContext.localizedFallbackTitle = localizedFallbackTitle;
+          }
 
+          if #available(iOS 9.0, *) {
+              policy = .deviceOwnerAuthentication;
```
Comment (ajnrajeev) [inline src/ios/Fingerprint.swift]: The issue with this piece of code is that in ios 9+ devices if you don't set the authenticationContext.localizedFallbackTitle as '', iOS will show 'Enter Passcode' button on non-matching finger print, but on clicking there will be no action. The purpose of the initial commit was to fix this issue. Also authenticationContext.localizedFallbackTitle set to '' is enough for disable backup in ios9+ devices. This basically removed the 'Enter Passcode' button from the native popup. So I also suggest that disable backup and policy should be set separately, how this was before

Comment (ajnrajeev) [inline src/ios/Fingerprint.swift]: I have fixed and push on top.

Comment (NiklasMerz) [inline src/ios/Fingerprint.swift]: Thanks for clarifying this. I was not aware of this behavior of the iOS API. Setting the policy should be more straightforward.

**Comment by NiklasMerz** [comment]
@ajnrajeev Thanks again for contributing. Your changes are now in master. 

If you add your git E-mail-address to your Github profile you will be listed as a contributor.

**Pull request merged**